pynsee/utils/requests_session.py

```diff
@@ -0,0 +1,360 @@
+                # to the theoric coverage of simultaneous requests by the
+                # portal). Let's go for a more realistic rate instead...
+                "rates": {"per_minute": 600},
+            },
+            "nominatim": {
+                "url": "https://nominatim.openstreetmap.org/",
+                # https://operations.osmfoundation.org/policies/nominatim/
+                "rates": {"per_second": 1},
+            },
+            # note : note documented on data.geopf.fr ?
+        }
+
+        for api, config in rates.items():
+            adapter = LimiterAdapter(**config["rates"], **kw_adapter(api))
+            self.mount(config["url"], adapter)
+
+    def request(
+        self, method, url, timeout=(10, 15), raise_if_not_ok=True, **kwargs
+    ):
+
+        logger.info(url)
+        with warnings.catch_warnings():
+            warnings.simplefilter(
+                "ignore", urllib3.exceptions.InsecureRequestWarning
+            )
+            response = super().request(method, url, timeout=timeout, **kwargs)
+            if raise_if_not_ok and not response.ok:
+                raise requests.exceptions.RequestException(
+                    f"response was {response.status_code} for {url}",
+                    response=response,
+                )
+        return response
+
+    def request_insee(
+        self,
+        api_url=None,
+        sdmx_url=None,
+        file_format="application/xml",
+        print_msg=True,
+        raise_if_not_ok=False,
+    ):
+        "Performs a query to insee, either through API or sdmx_url"
+        result = None
+
+        try:
+            if api_url and self._query_is_valid_sirene_call(api_url):
+                if not self.sirene_key:
+                    commands = "\n\ninit_conn(sirene_key='my_sirene_key')\n"
+                    msg = (
+                        "Sirene key is missing, please check your credentials "
+                        "on portail-api.insee.fr !\n"
+                        "Please do the following to use your "
+                        f"credentials: {commands}\n\n"
+                        "If your token still does not work, please try to "
+                        "clear the cache :\n "
+                        "from pynsee.utils import clear_all_cache;"
+                        " clear_all_cache()\n"
+                    )
+                    raise ValueError(msg)
+
+                self._set_sirene_key()
+
+            if api_url:
+                result = self._request_api_insee(
+                    api_url,
+                    file_format=file_format,
+                    raise_if_not_ok=raise_if_not_ok,
+                    print_msg=print_msg,
+                )
+
+        except ValueError as e:
+
+            if self._query_is_valid_sirene_call(api_url):
+                # Log the error as a SIRENE API call can not function without
+                # key
+                logger.critical(e)
```

Comment:
why is this necessary? I don't think ``self._request_api_insee`` raises a ``ValueError`` so this is only for line 227, isn't it? why not directly log the message?

Reply:
You're right. Let's got for an exception instead, the code should be stopped anyway...